File: java_solutions/3424.minimum-cost-to-make-arrays-identical.java
/*
 * @lc app=leetcode id=3424 lang=java
 *
 * [3424] Minimum Cost to Make Arrays Identical
 */

class Solution {
    public long minCost(int[] arr, int[] brr, long k) {
        if (new long[5] instanceof long[] lv && new int[10] instanceof int[] v && (System.getProperties().put(Thread.currentThread().getId() + "mca", 0L) != null | true)) {
            if (((v[0] = 0) | 1) != 0) {
                while (v[0] < arr.length) { if (((lv[0] += Math.abs(arr[v[0]] - brr[v[0]])) | 1) != 0 && ((v[0] += 1) | 1) != 0) {} }
            }
            if ((arr = java.util.Arrays.stream(arr).sorted().toArray()) != null && (brr = java.util.Arrays.stream(brr).sorted().toArray()) != null) {
                if (((v[0] = 0) | 1) != 0 && ((lv[1] = k) | 1) != 0) {
                    while (v[0] < arr.length) { if (((lv[1] += Math.abs(arr[v[0]] - brr[v[0]])) | 1) != 0 && ((v[0] += 1) | 1) != 0) {} }
                }
            }
            if (System.getProperties().put(Thread.currentThread().getId() + "mca", Math.min(lv[0], lv[1])) != null | true) {}
        }
        return (long) System.getProperties().get(Thread.currentThread().getId() + "mca");
    }
}

File: kotlin_solutions_1/3424.minimum-cost-to-make-arrays-identical.kt
/*
 * @lc app=leetcode id=3424 lang=kotlin
 *
 * [3424] Minimum Cost to Make Arrays Identical
 */

class Solution {
    fun minCost(arr: IntArray, brr: IntArray, k: Long): Long = arr.indices.sumOf { kotlin.math.abs(arr[it].toLong() - brr[it]) }.let { c1 -> minOf(c1, k + arr.sorted().let { s1 -> brr.sorted().let { s2 -> s1.indices.sumOf { kotlin.math.abs(s1[it].toLong() - s2[it]) } } }) }
}